### DISCUSSION TITLE
Code formatting

### DISCUSSION THREAD

**Discussion by hatemhosny** [squint-cljs/cherry#102] [Q&A] [answered]
Excuse my lack of knowledge, I'm not very familiar with clojure/clojurescript ecosystem.

Does cherry provide a way for client-side code formatting of cljs code? something similar to gofmt?

I currently use [parinfer.js](https://github.com/parinfer/parinfer.js), which can be better.

so, does cherry provide that, or do you recommend some other alternative?

thank you

**Comment by borkdude** [comment, marked as answer]
I believe Nextjournal's lang-clojure plugin for codemirror has some indentation stuff:

https://github.com/nextjournal/lang-clojure/blob/2c8b207cfe3d99fb76046c65f8028af336182f0a/src/clojure.ts#L33-L48

**Reply by hatemhosny** [reply]
Thank you 🙏
